crates/router/src/core/payments/operations/payment_create.rs

```diff
@@ -399,12 +402,13 @@ impl<F: Send + Clone> GetTracker<F, PaymentData<F>, api::PaymentsRequest> for Pa
                             api_models::payments::MandateIds {
                                 mandate_id: Some(mandate_obj.mandate_id),
                                 mandate_reference_id: Some(api_models::payments::MandateReferenceId::ConnectorMandateId(
-                                api_models::payments::ConnectorMandateReferenceId{
-                                    connector_mandate_id: connector_id.connector_mandate_id,
-                                    payment_method_id: connector_id.payment_method_id,
-                                    update_history: None,
-                                    mandate_metadata: None,
-                                }
+                                api_models::payments::ConnectorMandateReferenceId::new(
+                                    connector_id.get_connector_mandate_id(),
+                                    connector_id.get_payment_method_id(),
+                                    None,
+                                    None,
+                                    None
```

Comment:
can we populate connector_mandate_request_reference_id here?

Reply:
added